templates/monorepo/README.md

```diff
@@ -0,0 +1,82 @@
+# TypeScript Monorepo Template
+
+Scaffold a new project using a monorepo structure.
+Each monorepo is meant to include all the published artifacts for the project as well as the infrastructure definition.
+
+## Tasks
+
+Tasks are defined in the `turbo.json` and `package.json` files. To execute a task, just run the command at the project root:
+
+```sh
+yarn <cmd>
+```
+
+`Turborepo` will execute the task for all the workspaces that declare the same command in their `package.json` file; it also applies caching policies to the command according to the rules defined in `turbo.json`.
+
+To define a new task:
+
+- add the definition to `turbo.json` under `pipeline`;
+- add a script with the same name in `package.json` as `turbo <cmd name>`.
+
+Defined tasks are _lint_, _test_, and _typecheck_.
+
+## Dependencies
+```sh
+# install all dependencies for the project
+yarn
+
+# install a dependency to a workspace
+#   (workspace name is the name in the package.json file)
+yarn workspace <workspace name> add <package name>
+yarn workspace <workspace name> add -D <package name>
+
+# install a dependency for the monorepo
+#   (ideally a shared dev dependency)
+yarn add -D <package name>
+```
+
+To add a dependency to a local workspace, manually edit the target workspace's `package.json` file adding the dependency as
+
+```json
+"dependencies": {
+    "my-dependency-workspace": "workspace:*"
+}
+```
+
+## Folder structure
+
+### `/apps`
+
+It contains the applications included in the project.
+Each folder is meant to produce a deployable artifact; how and where to deploy it is demanded to a single application.
+
+Each sub-folder is a workspace.
+
+### `/packages`
+
+Packages are reusable TypeScript modules that implement a specific logic of the project. They are meant for sharing implementations across other apps and packages of the same projects, as well as being published in public registries.
+
+Packages that are meant for internal code sharing have `private: true` in their `package.json` file; all the others are meant to be published into the public registry.
+
+Each sub-folder is a workspace.
+
+### `/infra`
+It contains the _infrastructure-as-code_ project that defines the resources for the project as well as the executuion environments. Database schemas and migrations are defined here too, in case they are needed.
+
+### `/configs`
```

Comment:
They are technically the same, and Turbo itself does that. I think having dedicated folders will result in better readability as you don't usually need to look at that file, but you'll probably want to navigate the `/packages` folder tree frequently.

Reply:
I agree to move them into packages, that's where they're usually found.

Reply:
Well, _usually_ they are at project's root 🤗

Reply:
Let's compare both scenarios, so I can show you how I reasoned about it.

**Scenario A: configs in `/configs`**

The complete template code (that is: the codebase we develop in) would look like:
```
monorepo/
├─ configs/
│  ├─ typescript-config-react/
│  ├─ typescript-config-react-native/
│  ├─ typescript-config-next/
│  ├─ typescript-config-node/
│  ├─ eslint-config-react/
│  ├─ eslint-config-react-native/
│  ├─ eslint-config-next/
│  ├─ eslint-config-node/
├─ apps/
|  ├─ docs/
│  ├─ react-app/
│  ├─ react-native-app/
│  ├─ next-app/
│  ├─ node-app/
│  ├─ azure-functions-app/
│  ├─ dapr-app/
├─ packages/
│  ├─ node-package1/
│  ├─ node-package2/
│  ├─ node-package3/
│  ├─ node-package3/
│  ├─ react-package1/
│  ├─ react-package2/
│  ├─ react-package3/
├─ tests/
│  ├─ cypress-e2e/
│  ├─ detox-e2e/
│  ├─ k6-load/
│  ├─ k6-stress/
│  ├─ rest-api-test/
```

Pretending to image a typical user instance of such template, it would be
```
monorepo/
├─ configs/
│  ├─ typescript-config-react/
│  ├─ typescript-config-node/
│  ├─ eslint-config-react/
│  ├─ eslint-config-node/
├─ apps/
|  ├─ docs/
│  ├─ react-app/
│  ├─ azure-functions-app1/
│  ├─ azure-functions-app2/
├─ packages/
│  ├─ models/
│  ├─ ui/
│  ├─ shared-utils/
├─ tests/
│  ├─ cypress-e2e/
│  ├─ rest-api-test/
```

**Scenario B: configs in `/packages`**

Same as above, but:
```
monorepo/
├─ apps/
|  ├─ docs/
│  ├─ react-app/
│  ├─ react-native-app/
│  ├─ next-app/
│  ├─ node-app/
│  ├─ azure-functions-app/
│  ├─ dapr-app/
├─ packages/
│  ├─ eslint-config-react/
│  ├─ eslint-config-react-native/
│  ├─ eslint-config-next/
│  ├─ eslint-config-node/
│  ├─ node-package1/
│  ├─ node-package2/
│  ├─ node-package3/
│  ├─ node-package3/
│  ├─ react-package1/
│  ├─ react-package2/
│  ├─ react-package3/
│  ├─ typescript-config-react/
│  ├─ typescript-config-react-native/
│  ├─ typescript-config-next/
│  ├─ typescript-config-node/
├─ tests/
│  ├─ cypress-e2e/
│  ├─ detox-e2e/
│  ├─ k6-load/
│  ├─ k6-stress/
│  ├─ rest-api-test/
```

and for the user
```
monorepo/
├─ apps/
|  ├─ docs/
│  ├─ react-app/
│  ├─ azure-functions-app1/
│  ├─ azure-functions-app2/
├─ packages/
│  ├─ eslint-config-react/
│  ├─ eslint-config-node/
│  ├─ models/
│  ├─ ui/
│  ├─ shared-utils/
│  ├─ typescript-config-react/
│  ├─ typescript-config-node/
├─ tests/
│  ├─ cypress-e2e/
│  ├─ rest-api-test/
```

Reply:
ok, with these examples I am 100% convinced that we should NOT include these packages (always with the same content) inside the monorepos but publish them on NPM and make them workspace dependencies

Reply:
I agree, let's just _mitigate_ the features we are losing.

By having a single config file (per ws type) we enforce the same version by design (you cannot have two different react versions).

Maybe we can use yarn constraints for that https://yarnpkg.com/features/constraints

Reply:
do you (both) think it's worth to postpone the package publication - and continue working in this playground - until we reach a more stable checkpoint?

Reply:
yep! so we can include these configuration in workspace template instead the monorepo one (since they are not needed now)